File: src/main/java/app/askresume/domain/resume/constant/DifficultyType.java
package app.askresume.domain.resume.constant;

import lombok.Getter;

import java.util.Arrays;
import java.util.List;

@Getter
public enum DifficultyType {

    EASY("easy"), MEDIUM("medium"), HARD("hard");

    private final String type;

    public String type() {
        return type;
    }

    DifficultyType(String type) {
        this.type = type;
    }

    public static boolean isDifficultyType(String type) {
        List<DifficultyType> difficultyTypes = Arrays.stream(DifficultyType.values())
                .filter(difficultyType -> difficultyType.name().equalsIgnoreCase(type)).toList();
        return difficultyTypes.size() != 0;
    }


}

File: src/main/kotlin/app/askresume/domain/generative/interview/constant/DifficultyType.kt
package app.askresume.domain.generative.interview.constant

import java.util.*


enum class DifficultyType(
    private val value: String
) {

    EASY("easy"),
    MEDIUM("medium"),
    HARD("hard")
    ;

    fun value() = value

    companion object {
        fun isDifficultyType(type: String): Boolean {
            val difficultyTypes = Arrays.stream(values())
                .filter { difficultyType -> difficultyType.name.equals(type, ignoreCase = true) }.toList()
            return difficultyTypes.size != 0
        }
    }
}